### PR TITLE
runtime: don't execute BeforeAll and AfterAll hooks when in dry-run

### LINKED ISSUE #1258: BeforeAll hook executes in dry-run mode
### Problem
I have a setup with cucumber and webdriverio and I use a BeforeAll hook to connect to the webdriver server.
I also have a command that runs cucumber in dry-run mode just to print out snippets for the missing step definitions. But when I run that command, the BeforeAll hook still executes and it tries to connect to the webdriver server, which of course fails if I'm not running the server and I never get to see my snippets.

### Replicate
First create these two files
**package.json**
```json
{
  "scripts": {
    "start": "cucumber-js --dry-run --require hooks.js"
  },
  "devDependencies": {
    "cucumber": "^6.0.5"
  }
}
```
**hooks.js**
```js
const BeforeAll = require('cucumber').BeforeAll;
BeforeAll(function() {
  console.log('BeforeAll executes');
});
```
Then run these two commands
```
$ npm install
...
$ npm start -s
BeforeAll executes


0 scenarios
0 steps
0m00.000s
```
I would expect the BeforeAll hook to never execute

### Environment
|Software|Version|
|-|-|
| Ubuntu | 18.04.2|
|npm|6.12.0|
|cucumber-js|6.0.5|

I get the same result with `yarn 1.19.1`

### Workaround
In this case connecting to the webdriver server in the world constructor does not work (or not very well at least) since the world constructor is strictly synchronous and I want to run webdriverio in async mode.
My best workaround is setting an environment variable in the script command and check if this is set in the hook and just return if it is.

### PR DESCRIPTION
**Pull request by davidjgoss**
Fixes #1258

### CODE REVIEW

File: features/dryrun_mode.feature [hunk 1]
```diff
@@ -1,3 +1,4 @@
+@dryrun
```
Comment (charlierudolph) [inline features/dryrun_mode.feature]: Do we need this? You should be able to run this feature only by targeting the file

File: src/runtime/index.ts [hunk 2]
```diff
@@ -61,6 +61,9 @@ export default class Runtime {
   }
 
   async runTestRunHooks(key: string, name: string): Promise<void> {
+    if (this.options.dryRun) {
+      return
+    }
```
Comment (charlierudolph) [inline src/runtime/index.ts]: We may need this in parallel runtime as well

File: features/dryrun_mode.feature (hunk 1 above)
Comment (davidjgoss) [reply features/dryrun_mode.feature]: Yeah, true, removed

File: src/runtime/index.ts (hunk 2 above)
Comment (davidjgoss) [reply src/runtime/index.ts]: Ahh totally missed that! Pushed changes now that cover it.

File: features/dryrun_mode.feature [hunk 3]
```diff
@@ -37,3 +37,36 @@ Feature: Dryrun mode
     When I run cucumber-js with `--dry-run`
     Then it fails
     And scenario "some scenario" step "Given a step" has status "undefined"
+
+  Scenario: hooks should not execute in dry run, single runtime
```
Comment (charlierudolph) [inline features/dryrun_mode.feature]: single -> serial?

Comment (davidjgoss) [reply features/dryrun_mode.feature]: sold!

**Pull request merged**